R/regression.R

```diff
@@ -618,46 +637,102 @@ importanceFooter <- function(x)
 #'   coerced to that class): a symbolic description of the model to be fitted.
 #'   The details of type specification are given under \sQuote{Details}.
 #' @param .estimation.data A \code{\link{data.frame}}.
-#' @param subset Not used.
 #' @param .weights An optional vector of sampling weights, or, the name or, the
 #'   name of a variable in \code{data}. It may not be an expression.
 #' @param type See \link{Regression}.
 #' @param robust.se Computes standard errors that are robust to violations of
 #'   the assumption of constant variance, using the HC1 (degrees of freedom)
 #'   modification of White's (1980) estimator (Long and Ervin, 2000). This parameter is ignored
 #'   if weights are applied (as weights already employ a sandwich estimator).
+#' @param outlier.prop.to.remove A single numeric value that determines the percentage of data points to remove from the
+#'   analysis. The data points removed correspond to those in the proportion with the largest residuals.
+#'   A value of 0 or NULL would denote no points are removed. A value x, with 0 < x < 0.5 (not inclusive) would
+#'   denote that a percentage between none and 50\% of the data points are removed.
+#' @param seed A integer seed to generate the surrogate residuals.
 #' @param ... Arguments to the wrapped functions.
 #' @importFrom flipData CalibrateWeight WeightedSurveyDesign
+#' @importFrom flipFormat FormatAsPercent
 #' @importFrom flipU InterceptExceptions
 #' @importFrom MASS polr glm.nb
 #' @importFrom nnet multinom
 #' @importFrom stats glm lm poisson quasipoisson binomial pt quasibinomial
 #' @importFrom survey svyglm svyolr
 #' @export
-FitRegression <- function(.formula, .estimation.data, subset, .weights, type, robust.se, ...)
+FitRegression <- function(.formula, .estimation.data, .weights, type, robust.se, outlier.prop.to.remove, seed = 12321, ...)
 {
-    weights <- .weights #Does nothing, except remove notes from package check.
     .design <- NULL
+    # Initially fit model on all the data and then refit with outlier removal if necessary
+    fitted.model <- fitModel(.formula, .estimation.data, .weights, type,
+                             robust.se, subset = NULL, ...)
+    model <- fitted.model$model
+    .design <- fitted.model$design
+    .estimation.data <- fitted.model$estimation.data
+    .formula <- fitted.model$formula
+    remove.outliers <- checkAutomaterOutlierRemovalSetting(outlier.prop.to.remove)
+    if (remove.outliers)
+    {
+        # Don't support Multinomial Logit for now.
+        if (type == "Multinomial Logit")
+            stop("Automated outlier removal and re-fitting a 'Multinomial Logit' model is not supported")
```

Comment:
what about instead warning and continuing without the outlier removal in the multinomial case?

Reply:
I agree that is better. I'll amend the code in a new commit to do this.